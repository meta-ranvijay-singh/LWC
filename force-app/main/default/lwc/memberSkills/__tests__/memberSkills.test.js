import { getCreateComponent, flushPromises } from "c/test_utils";
import MemberSkills from "c/memberSkills";

const createComponent = getCreateComponent(MemberSkills, "c-member-skills");

const TeamsMockData = require("./data/teamsData.json");

describe("c-member-skills", () => {
  afterEach(() => {
    // The jsdom instance is shared across test cases in a single file so reset the DOM
    while (document.body.firstChild) {
      document.body.removeChild(document.body.firstChild);
    }
  });

  describe("Create component", () => {
    it("renders component correctly", () => {
      // When
      const { component } = createComponent();

      // Then
      expect(component).toMatchSnapshot();
    });
  });

  it("verify data insert on submit button click", async () => {
    // Create initial element
    const { root, component } = createComponent({ teams: TeamsMockData });

    let inputField = root.querySelectorAll("lightning-input");

    let name = inputField[0];
    name.value = "Tom";
    name.dispatchEvent(new CustomEvent("change"));

    let skill = inputField[1];
    skill.value = "Batsman";
    skill.dispatchEvent(new CustomEvent("change"));

    let combo = root.querySelectorAll("lightning-combobox")[0];
    combo.value = TeamsMockData[0].Id;
    combo.dispatchEvent(new CustomEvent("change"));

    const submitButton = root.querySelectorAll("lightning-button")[0];

    const handler = jest.fn();
    component.addEventListener("insert", handler);

    submitButton.dispatchEvent(new CustomEvent("click"));

    await flushPromises();
    expect(handler).toHaveBeenCalledTimes(1);
  });
});
