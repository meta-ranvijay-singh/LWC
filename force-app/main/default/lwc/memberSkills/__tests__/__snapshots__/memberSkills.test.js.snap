// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`c-member-skills Create component renders component correctly 1`] = `
<c-member-skills>
  #shadow-root(open)
    <lightning-card
      class="slds-size_medium slds-p-around_medium"
    >
      #shadow-root(open)
        <slot />
        <slot
          name="actions"
        />
        <slot
          name="footer"
        />
        <slot
          name="title"
        />
      <lightning-input
        class="slds-size_1-of-3 slds-align_absolute-center"
      >
        #shadow-root(open)
      </lightning-input>
      <lightning-combobox
        class="slds-size_1-of-3 slds-align_absolute-center"
      >
        #shadow-root(open)
      </lightning-combobox>
      <lightning-input
        class="slds-size_1-of-3 slds-align_absolute-center"
      >
        #shadow-root(open)
      </lightning-input>
      <div
        slot="footer"
      >
        <lightning-button
          class="slds-m-left_x-small"
          title="Primary action"
        >
          #shadow-root(open)
        </lightning-button>
      </div>
    </lightning-card>
</c-member-skills>
`;
